src/libraries/Common/src/System/Security/Cryptography/SLHDsaAlgorithm.cs

```diff
@@ -20,112 +22,215 @@ internal sealed class SLHDsaAlgorithm
         public string Name { get; }
 
         /// <summary>
-        ///  Initializes a new instance of the <see cref="SLHDsaAlgorithm" /> structure with a custom name.
+        ///  Gets the size of the secret key in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the secret key in bytes for this algorithm.
+        /// </value>
+        public int SecretKeySizeInBytes { get; }
+
+        /// <summary>
+        ///  Gets the size of the public key in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the public key in bytes for this algorithm.
+        /// </value>
+        public int PublicKeySizeInBytes { get; }
+
+        /// <summary>
+        ///  Gets the size of the signature in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the signature in bytes for this algorithm.
+        /// </value>
+        public int SignatureSizeInBytes { get; }
+        public string Oid { get; }
```

Comment:
OID should be internal... and be separated from a property with xmldoc.

```suggestion

        internal string Oid { get; }
```